lib/vmpooler/vsphere_helper.rb

```diff
@@ -6,14 +6,23 @@ class VsphereHelper
     DISK_TYPE = 'thin'
     DISK_MODE = 'persistent'
 
-    def initialize(_vInfo = {})
-      config_file = File.expand_path('vmpooler.yaml')
-      vsphere = YAML.load_file(config_file)[:vsphere]
-
-      @connection = RbVmomi::VIM.connect host: vsphere['server'],
-                                         user: vsphere['username'],
-                                         password: vsphere['password'],
-                                         insecure: true
+    def initialize(credentials)
```

Comment:
One benefit to this change is that connections are no longer made to vsphere when a thread is initialized. Instead, they are created when work is performed. The result is the application and its threads are available faster, and we do not open as many connections as pools when the application starts.